packages/tpetra/core/test/FECrsMatrix/FECrsMatrix_UnitTests.cpp

```diff
@@ -367,14 +370,17 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( FECrsMatrix, Assemble1D_2, LO, GO, Scalar, No
   if (fe_matrix.getRowMap()->isNodeLocalElement(local_row))
   {
     auto num_entries = fe_matrix.getNodeNumEntries();
-    Teuchos::Array<LO> cols(num_entries);
+    cols_type cols("cols",num_entries);
     Scalar zero = Teuchos::ScalarTraits<Scalar>::zero();
     Scalar one = Teuchos::ScalarTraits<Scalar>::one();
-    Teuchos::Array<Scalar> vals(num_entries, zero);
-    fe_matrix.getLocalRowCopy(local_row, cols(), vals(), num_entries);
-    std::fill(vals.begin(), vals.end(), zero);
+    vals_type vals("vals",num_entries);
+    fe_matrix.getLocalRowCopy(local_row, cols, vals, num_entries);
+    Kokkos::deep_copy(vals,zero);
```

Comment:
Doesn't Kokkos zero out by default?

Reply:
The deep_copy afterwards is to zero the row after we get it